### ISSUE TITLE
Problems with HeartBeat / RawMessage

### ISSUE THREAD

**Issue by leandrogomes777** [connamara/quickfixn#779]
Hi, we migrate from an older version of QuickFix (that one that was an wrapper for C++) to the latest one (1.10.0), and we have some  applications using RawMessages.

We are facing problems with HeartBeat and RawMessages.
When we are sending the message (that use the method below), the LastSentTimeDT from "state" object is not being updated, so we are originating heartbeats while sending and receiving messages.

```
public bool Send(string message)
{
      lock (sync_)
      {
          if (null == responder_)
              return false;
          this.Log.OnOutgoing(message);
          return responder_.Send(message);
      }
  }
```

Could you check please?

Thanks,
Leandro

**Comment by mgatny** [comment]
I'm not sure why the behavior would differ from the C++ wrapper.  QF/c++ does the same thing that QF/n does: it only updates the SessionState's last-sent-time when initializing the Header before sending (in QF/c++ this method is called `fill`, in QF/n it is `InitializeHeader`).

**Comment by leandrogomes777** [comment]
@mgatny
I've checked the C++ version and you are completely right, the behavior is exactly the same. But the method is private in the C++ and not exists in the wrapper (1.13.3).
I will verify, because we've made changes on the original quickfix in ower application for performance reasons and we've made the method public in the C++ and in Quickfix/N is already public.
We've made a workaround in ower appplication to update the variable using reflection. 
Fell free to close the ticket.

Thanks,
Leandro
